modules/standard/Types.chpl

```diff
@@ -596,30 +594,25 @@ iter type enumerated.these(){
     yield i;
 }
 
-// TODO add chpl_ to these functions' names - they are not intended for user.
-private proc enum_minbits(type t: enumerated) param {
-  return __primitive( "enum min bits", t);
+private proc chpl_enum_minbits(type t: enumerated) param {
+  if t.size <= max(uint(8)) then
+    return 8;
+  if t.size <= max(uint(16)) then
+    return 16;
+  if t.size <= max(uint(32)) then
+    return 32;
+  return 64;
```

Comment:
It's going to be a bad day for the compiler when somebody makes an enum with 2**63 constants.

Reply:
I'm not sure how to interpret this.  Are you seeing a bug / limitation that I'm not?

Reply:
No, I'm just saying that the compiler theoretically supports it but one will run out of memory and time before it would ever matter.